### ISSUE TITLE
Error installing wheel with "invalid version"

### ISSUE THREAD

**Issue by albertotb** [astral-sh/uv#5478]
Version: `uv 0.2.29`

I'm not sure if this is intended or not:

```
uv pip install https://huggingface.co/PlanTL-GOB-ES/es_anonimization_core_lg/resolve/main/es_anonimization_core_lg-any-py3-none-any.whl
```

Fails with

```
error: The wheel filename "es_anonimization_core_lg-any-py3-none-any.whl" has an invalid version: expected version to start with a number, but no leading ASCII digits were found
```

However, the equivalent pip command is able to install the wheel:

```
pip install https://huggingface.co/PlanTL-GOB-ES/es_anonimization_core_lg/resolve/main/es_anonimization_core_lg-any-py3-none-any.whl
Collecting es-anonimization-core-lg==any
  Downloading https://huggingface.co/PlanTL-GOB-ES/es_anonimization_core_lg/resolve/main/es_anonimization_core_lg-any-py3-none-any.whl (587.8 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 587.8/587.8 MB 1.5 MB/s eta 0:00:00
```

**Comment by charliermarsh** [comment]
In our current design this is intentional -- we require that wheels always have valid wheel filenames, and that isn't a valid filename (since it's omitting the version).

**Comment by charliermarsh** [comment]
We should support this, requires some refactors though.

**Comment by eth3lbert** [comment]
> Version: `uv 0.2.29`
> 
> I'm not sure if this is intended or not:
> 
> ```
> uv pip install https://huggingface.co/PlanTL-GOB-ES/es_anonimization_core_lg/resolve/main/es_anonimization_core_lg-any-py3-none-any.whl
> ```
> 
> Fails with
> 
> ```
> error: The wheel filename "es_anonimization_core_lg-any-py3-none-any.whl" has an invalid version: expected version to start with a number, but no leading ASCII digits were found
> ```
> 
> However, the equivalent pip command is able to install the wheel:
> 
> ```
> pip install https://huggingface.co/PlanTL-GOB-ES/es_anonimization_core_lg/resolve/main/es_anonimization_core_lg-any-py3-none-any.whl
> Collecting es-anonimization-core-lg==any
>   Downloading https://huggingface.co/PlanTL-GOB-ES/es_anonimization_core_lg/resolve/main/es_anonimization_core_lg-any-py3-none-any.whl (587.8 MB)
>      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 587.8/587.8 MB 1.5 MB/s eta 0:00:00
> ```

Apologies for the intrusion. Could you please specify which version of `pip` you are using?

I have tested this with `pip` version `24.2` and encountered the following error:
```shell
pip install https://huggingface.co/PlanTL-GOB-ES/es_anonimization_core_lg/resolve/main/es_anonimization_core_lg-any-py3-none-any.whl               
ERROR: Invalid requirement: 'es-anonimization-core-lg==any': Expected end or semicolon (after name and no valid version specifier)
    es-anonimization-core-lg==any
```

**Comment by albertotb** [comment]
Sure, I'm using pip 24.0, which seems to work fine. I will report to the creator of the wheel to see if they can follow the naming conventions. The confusing part here is that with pip seems to work (at least in some versions), so the wheel mantainers may not have noticed that the naming is incorrect.

**Comment by albertotb** [comment]
Looking at pip release notes maybe this is related to this issue https://github.com/pypa/pip/issues/12063

It seems that starting with pip 24.1b1 the wheel I used as an example is no longer installable with pip https://pip.pypa.io/en/stable/news/#b1-2024-05-06

**Comment by jcorrie** [comment]
I'm experiencing the same issue using uv pip to install a spacy model via hugging face. The url given by hugging face for the wheels file seems only (?) to provide "any" as the version number. Agreed with @albertotb that from the pip issue referenced it seems like this is probably something hugging face will need to update. I've open a new issue [huggingface/huggingface_hub#2637](https://github.com/huggingface/huggingface_hub/issues/2637)

**Comment by albertotb** [comment]
Yes, the workaround I'm using is to stick with pip to install those HuggingFace models and pin pip<24.0. I guess you could also host a clone of such HuggingFacel models but with the correct naming. That is not ideal if they are being updated regular, but it's probably fine for most models.

**Comment by jtauber** [comment]
can the wheel also be downloaded and renamed?

**Comment by notatallshaw** [comment]
> We should support this, requires some refactors though.

Does uv have a strong reason to support this other than it happens to work with pip?

Pip is moving towards only allowing valid wheels, in fact this was going to be broken by https://github.com/pypa/pip/pull/12917, but as there was no deprication warning I'm going to push it out till pip 25.3: https://github.com/pypa/pip/pull/12917#issuecomment-2667222833, after which, like uv, pip will require wheels to have valid names in all cases.

**Comment by charliermarsh** [comment]
Looking back at this, I agree with you that we shouldn't support it. Especially if pip is moving in that direction.
